cell_classification/segmentation_data_prep_test.py

```diff
@@ -168,5 +186,55 @@ def test_get_inst_binary_masks():
         assert np.array_equal(loaded_binary_img, instance_mask_eroded)
 
 
+def test_get_marker_activity():
+    data_prep = prep_object()
+    cell_table = prepare_cell_type_table()
+    conversion_matrix = prepare_conversion_matrix()
+    data_prep.cell_type_table = cell_table
+    marker = "CD11c"
+    sample_name = "fov_1"
+    fov_1_subset = cell_table[cell_table.SampleID == sample_name]
+    marker_activity = data_prep.get_marker_activity(
+        sample_name, conversion_matrix, marker
+    )
+
+    # check if the we get the correct marker activity
+    assert len(marker_activity) == len(fov_1_subset.label)
+
+    # check if the out_dict has the right marker activity values for a given cell
+    for i in range(len(fov_1_subset.label)):
+        assert (
+            marker_activity[i] == conversion_matrix.loc[fov_1_subset.cluster_labels[i], "CD11c"]
+        )
+
+
+def test_get_marker_activity_mask():
+    data_prep = prep_object()
+    marker = "CD11c"
+    marker_activity = [1, 0, 0, 0, 0, 1]
+    instance_mask = np.zeros([256, 256], dtype=np.uint16)
+    instance_mask[0:32, 0:32] = 1
+    instance_mask[0:32, 32:64] = 2
+    instance_mask[0:32, 64:96] = 3
+    instance_mask[32:64, 0:32] = 4
+    instance_mask[64:96, 64:96] = 5
+    instance_mask[128:160, 128:160] = 6
+    binary_mask = (instance_mask > 0).astype(np.uint8)
+
+    marker_activity_mask = data_prep.get_marker_activity_mask(
+        instance_mask, binary_mask, marker_activity
+    )
+
+    # check if the right spatial dimensions got returned
+    assert marker_activity_mask.shape == instance_mask.shape
+
+    # check if the right marker activity values are returned
+    instance_mask[binary_mask < 1] = 0
```

Comment:
What is the point of this line?

Reply:
Was there because I tested the function with the real eroded binary mask in the first place. I changed this to a simple non-eroded binary mask and thus don't need this line anymore.